app/controllers/api/v2/courses_controller.rb

```diff
@@ -97,6 +101,17 @@ def update_sites
         @course.errors[:sites] << "^You must choose at least one location" if site_ids.empty?
       end
 
+      def update_site_statuses
+        return if @course.errors.any?
+        return unless @study_mode_changed
```

Comment:
Minor suggestion: can we just use `@course.study_mode_changed?` here? I think it's literal enough and it seems a bit tricky that this is set in `update_course` and relies on that running before this runs (whereas `@course` is set in a before hook and is a bit more predictable).

Reply:
`@course.study_mode_changed?` resets after the `@course.save` call.

Reply:
🤦‍♂

Reply:
@davidgisbey I knew there was a magic thing you could of used `study_mode_changed?`

Reply:
Docs: https://edgeapi.rubyonrails.org/classes/ActiveModel/Dirty.html

Reply:
I guess you could do `@course.study_mode_previously_changed?` but it makes me think "when did it previously change?" 🤷‍♂

Reply:
That's an interesting one @richrace . Since it's on the `@course` object, it would have to be within the same request cycle, and since this is an update request that could actually make sense.

Reply:
Actually, this ties in with a bit of cognitive dissonance I'm experiencing with `update_site_statuses` in general. I feel like in `update` we should be calling:

```
      if @course.study_mode_previously_changed?
        update_site_status_vac_statuses
      end
```

... it just wasn't at all clear to me that the goal of this method is to update the `vac_status` on each `site_status`, if the course's `study_mode` has been changed. I think that should be made clearer.

Reply:
Have updated it to Misha's suggestion.